### PR TITLE
federation-apiserver test: Fixing the AfterEach order in e2e test

### PR DESCRIPTION
**Pull request by nikhiljindal**
Moving "It" and "AfterEach" into a "Describe" fixes the order of AfterEach.
Verified that now, test.AfterEach is run before running framework.AfterEach.

Removing the temporary disabling of framework.SkipUnlessFederated(f.Client) in AfterEach.

cc @kubernetes/sig-cluster-federation @mml

### CODE REVIEW

**Comment by lavalamp** [comment]
LGTM

File: test/e2e/federation-apiserver.go [hunk 1]
```diff
@@ -34,52 +34,54 @@ import (
...
-	AfterEach(func() {
-		// framework.SkipUnlessFederated(f.Client)  TODO: quinton-hoole: Temporarily disabled
-
-		// Delete registered clusters.
-		// This is if a test failed, it should not affect other tests.
-		clusterList, err := f.FederationClientset.Federation().Clusters().List(api.ListOptions{})
-		Expect(err).NotTo(HaveOccurred())
-		for _, cluster := range clusterList.Items {
-			err := f.FederationClientset.Federation().Clusters().Delete(cluster.Name, &api.DeleteOptions{})
+	Describe("testing clusters", func() {
```
Comment (quinton-hoole) [inline test/e2e/federation-apiserver.go]: nit: Reword this to be "Cluster objects" and the It() clause to "should be created and destroyed successfully".

**Comment by quinton-hoole** [comment]
@nikhiljindal One minor nit that could be addressed in a followup PR.

File: test/e2e/federation-apiserver.go (hunk 1 above)
Comment (nikhiljindal) [inline test/e2e/federation-apiserver.go]: Fixed

**Comment by nikhiljindal** [comment]
Fixed as per comment. Readding LGTM
